prompb/remote.proto

```diff
@@ -23,10 +23,32 @@ message WriteRequest {
   repeated prometheus.TimeSeries timeseries = 1 [(gogoproto.nullable) = false];
 }
 
+// ReadRequest represents a remote read request.
 message ReadRequest {
   repeated Query queries = 1;
+
+  enum ResponseType {
+    // Server will return a ReadResponse with raw samples.
+    //
+    // Response headers:
+    // Content-Type: "application/x-protobuf"
+    // Content-Encoding: "snappy"
+    SAMPLES = 0;
+
+    // Server will stream a varint delimited ChunkedReadResponse message that contains XOR encoded chunks for a single series.
+    //
+    // Response headers:
+    // Content-Type: "application/x-streamed-protobuf; proto=prometheus.ChunkedReadResponse"
+    // Content-Encoding: ""
+    STREAMED_XOR_CHUNKS = 1;
+  }
+  // response_type allows negotiating the content type of response.
+  // This does not guarantee the returned response. For servers that do not support this field, the SAMPLES response type
+  // is used.
+  ResponseType response_type = 2;
```

Comment:
> https://github.com/prometheus/prometheus/pull/5703#discussion_r296321539

@brian-brazil commented: 

> Shouldn't this be a list, in order of preference?

Yes more of "accepted" response types might make more sense, happy to address that.  I would propose
`repeated ResponseType accepted_response_types = 2;`

Reply:
I guess, if empty - we assume `[Samples]` though, right?

Reply:
Yes, that'd be a sane default.

Reply:
Cool, changing to that approach then (: